### PR TITLE
Add Value.Patch and Value.Find

### PR DESCRIPTION
**Pull request by dsnet**
This implements support for patching a JSON value according to RFC 6902
and looking up a value with a JSON Pointer (RFC 6901).

### CODE REVIEW

File: patch.go [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+			}
+		}
+		return nil
+	default:
+		return fmt.Errorf("hujson: patch operation %d: %v", i, err)
+	}
+}
+
+func (v *Value) patchRemoveOrReplace(i int, op patchOperation) error {
+	switch s, err := v.find(findState{pointer: op.path}); {
```
Comment (crawshaw) [inline patch.go:167]: I'm curious what led you to this style. It seems a little unusual. Go would typically read:

```
s, err := v.find(...)
if err != nil {
    return fmt.Errorf("...")
}
/// the err == nil case
```

File: patch.go [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+// the prefix must end on a path segment boundary.
+func hasPathPrefix(s, prefix string) bool {
+	if strings.HasPrefix(s, prefix) {
+		return len(s) == len(prefix) || s[len(prefix)] == '/'
+	}
+	return false
+}
+
+func equalValue(x, y Value) bool {
+	// TODO(dsnet): This definition of equality is both naive and slow.
```
Comment (crawshaw) [inline patch.go:225]: it's my favorite! :)

Comment (crawshaw) [inline patch.go:225]: it's my favorite! :)

**Review by crawshaw — APPROVED** [review]
I can't honestly say I understand all of the subtly of comment moving, but you do have a nice set of initial test cases. I also expect the only way we'd find issues with the heuristics will not be reading them, but seeing them in action. So, LGTM!

File: patch.go (hunk 1 above)
Comment (dsnet) [reply patch.go:167]: Now that I look at it, I'm not sure. At least I was consistent about my excessive usage of `switch` statements :)

I'll make it more conventional.

**Pull request merged**